sklearn/svm/_classes.py

```diff
@@ -189,6 +193,22 @@ class LinearSVC(LinearClassifierMixin, SparseCoefMixin, BaseEstimator):
     [1]
     """
 
+    _parameter_constraints = {
+        "penalty": StrOptions({"l1", "l2"}),
+        "loss": StrOptions({"hinge", "squared_hinge"}),
+        "dual": ["boolean"],
+        "tol": [Real],
+        "C": [Real],
+        "multi_class": StrOptions({"ovr", "crammer_singer"}),
+        "fit_intercept": ["boolean"],
+        "intercept_scaling": [Real],
+        "class_weight": [None, dict, "balanced"],
```

Comment:
Even if you want a single string option, e.g. `"balanced"` in this case, you'll still need to use `StrOptions` otherwise the validation function will try to interpret `"balanced"` as a special option such as `"boolean"`.

```suggestion
        "class_weight": [None, dict, StrOptions({"balanced"})],
```